### ISSUE TITLE
Increasing accuracy of imdb lstm example for network dataset

### ISSUE THREAD

**Issue by vinayakumarr** [tflearn/tflearn#210]
395216 train sequences X_train
98805 test sequences X_test
395216 class train sequences y_train
98805 class test sequences y_test

I have csv file which has 494021 rows and 42 columns: In that 41 are features and other one is class label: which has 23 classes.

I have used same imdb_lstm.py for my data set and my program is given below. I have tried with different network toplogies. It is giving training accuracy 40 and validation accuracy 56. I think something wrong in this and also i would like to increase the accuracy. How to do this?

from __future__ import print_function
import numpy as np
from sklearn.cross_validation import train_test_split
import tflearn
import pandas as pd
from tflearn.data_utils import to_categorical, pad_sequences

print("Loading")

data = pd.read_csv('NORM_sample.csv', header=None)

X = data.iloc[:,1:42]
y = data.iloc[:,0]

X_train, X_test, y_train, y_test = train_test_split(X, y,test_size=0.2,random_state=42)

print(X_train.shape)
print(X_test.shape)
print(y_train.shape)
print(y_test.shape)

print("Preprocessing")
X_train1 = X_train.values.T.tolist()
X_test1 = X_test.values.tolist()
y_train1 = y_train.values.T.tolist()
y_test1 = y_test.values.tolist()

# Sequence padding
trainX = pad_sequences(X_train1, maxlen=200, value=0.)
testX = pad_sequences(X_test1, maxlen=200, value=0.)

# Converting labels to binary vectors
trainY = to_categorical(y_train, nb_classes=0)
testY = to_categorical(y_test, nb_classes=0)

print(np.max(trainX)+1)

# Network building
net = tflearn.input_data([None, 200])
net = tflearn.embedding(net, input_dim=np.max(trainX)+1, output_dim=128)
net = tflearn.lstm(net, 128)
net = tflearn.dropout(net, 0.5)
net = tflearn.fully_connected(net, 24, activation='softmax')
net = tflearn.regression(net, optimizer='adam',loss='categorical_crossentropy')

# Training
model = tflearn.DNN(net, clip_gradients=0., tensorboard_verbose=0)
model.fit(trainX, trainY, validation_set=(testX, testY), show_metric=True, batch_size=128)

**Comment by aymericdamien** [comment]
You could try to use bidirectional_lstm:
```python
net = tflearn.bidirectional_lstm(tflearn.BasicLSTMCell(128), tflearn.BasicLSTMCell(128))
```

Also try to change LSTM by GRU, it may give you some improvements in certain cases.
